postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala

```diff
@@ -12,9 +13,21 @@ trait PostgresModule extends Jdbc { self =>
   override type TypeTagExtension[+A] = PostgresSpecific.PostgresTypeTag[A]
 
   object PostgresSpecific {
+    import self.ReadExecutor.DecodingError
     trait PostgresTypeTag[+A] extends Tag[A]
     object PostgresTypeTag {
-      implicit case object TInterval extends PostgresTypeTag[Interval]
+      implicit case object TInterval extends PostgresTypeTag[Interval] {
+        override def decode[Interval, DecodingError](
+          column: Either[Int, String],
+          resultSet: ResultSet
+        ): Either[DecodingError, Interval] =
+          scala.util
+            .Try(Interval.fromPgInterval(new PGInterval(column.fold(resultSet.getString(_), resultSet.getString(_)))))
+            .fold(
+              _ => Left(DecodingError.UnexpectedNull(column).asInstanceOf[DecodingError]),
```

Comment:
I had to add the `.asInstanceOf` to make the compiler happy with this

Reply:
Odd. I don't think this should be required, but the cast looks sound.